mage_ai/orchestration/pipeline_scheduler.py

```diff
@@ -183,6 +183,11 @@ def schedule(self, block_runs: List[BlockRun] = None) -> None:
         for b in self.pipeline_run.block_runs:
             b.refresh()
 
+        try:
+            self.__check_run_timeout()
```

Comment:
this method might not work with integration pipeline, streaming pipeline, and `run_pipeline_in_one_process` scenarios.

Reply:
does it make sense for a streaming pipeline to have timeouts? i think i can make it work for integration pipelines, but it will be tricky for the `run_pipeline_in_one_process` method for block run timeouts.

Reply:
no need to support it for the cases. just need to make sure we only run the check for the scenarios that we support.

Reply:
ok